app/jobs/spotlight/add_uploads_from_csv.rb

```diff
@@ -0,0 +1,37 @@
+##
+# Process a CSV upload into new Spotlight::Resource::Upload objects
+class Spotlight::AddUploadsFromCSV < ApplicationJob
+  queue_as :default
+  after_perform do |job|
+    csv_data, exhibit, user = job.arguments
+    Spotlight::IndexingCompleteMailer.documents_indexed(csv_data, exhibit, user).deliver_now
+  end
+
+  def perform(csv_data, exhibit, _user)
+    encoded_csv(csv_data).each do |row|
+      # The CSV row must have either a url or a file
+      url = row.delete('url')
+      file = row.delete('file')
+      next unless url.present? || file.present?
+
+      resource = Spotlight::Resources::Upload.new(
+        data: row,
+        exhibit: exhibit
+      )
+
+      url = "http://localhost:3000/#{file}" if file
+      resource.build_upload(remote_image_url: url)
+      resource.save_and_index
+    end
+  end
+
+  private
+
+    def encoded_csv(csv)
```

Comment:
Can we monkeypatch just the methods that we need to change, instead of copying the class over entirely?